[advaypakhale] iP

advaypakhale opened the pull request:
# Bob 🤓
---

> Bob is the best thing ever created! ~ Unknown Unbiased Reviewer

Bob helps you to organise your
* Todos
* Deadlines
*  Events

It's as easy as
1. Learn unix
2. Learn java
3. Run `java -jar bob.jar`

You can even spawn _multiple_ Bobs! Modify the main method:
```java
public static void main(String[] args) {
    Bob bob = new Bob();
    bob.run();
}
```

Bob can **transform** your life. Tick the following once it does:
- [ ] makes you more productive
- [ ] makes you less forgetful
- [ ] makes you see the world in each grain of sand

So what are you waiting for? Try Bob [today](https://time.is/)!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

advaypakhale committed:
docs: Add AI.md

advaypakhale committed:
refactor: Change duke to bob

advaypakhale committed:
feat: Add main loop logic for greet and exit

advaypakhale committed:
feat: Add main loop logic for greet and exit

advaypakhale committed:
feat: Add echo and bye logic

advaypakhale committed:
feat: Add list and add capability

advaypakhale committed:
refactor: Remove encapsulateSection from individual methods, centralise under processUserInput

advaypakhale committed:
fix: Make greet print from within method

advaypakhale committed:
feat: Add minimal Task class

advaypakhale committed:
feat: Add marking and unmarking functionality for tasks

advaypakhale committed:
feat: Add Deadline, Event, Todo classes

advaypakhale committed:
feat: Add todo, event, deadlines to chatbot

advaypakhale committed:
fix: Correct toString for Deadline class

advaypakhale committed:
test: Add text ui test

advaypakhale committed:
feat: Add error handling

advaypakhale committed:
feat: Add task deletion

advaypakhale committed:
docs: Modify AI.md

advaypakhale committed:
feat: Split into more classes following OOP

advaypakhale committed:
Merge branch 'branch-Level-7'

advaypakhale committed:
feat: Add datetime support

advaypakhale committed:
fix: Load from storage properly

advaypakhale committed:
Merge branch 'branch-Level-8'

advaypakhale committed:
feat: Split into packages

advaypakhale committed:
Merge branch 'branch-Level-8'

advaypakhale committed:
Merge branch 'add-gradle-support'

senshir commented on src/main/java/bob/command/CreateDeadlineCommand.java:
```diff
@@ -0,0 +1,52 @@
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException, IllegalCommandException {
+        String arguments = "";
```
arguments feels quite generic. maybe you could specify the arguments like dateArguments

senshir commented on src/main/java/bob/command/CreateDeadlineCommand.java:
```diff
@@ -0,0 +1,52 @@
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException, IllegalCommandException {
+        String arguments = "";
+        for (int i = 1; i < userInput.length; i++) {
+            arguments += userInput[i] + " ";
```
String concatenation with addition of " " could sometimes be inefficient. Maybe try using StringBuilder?

senshir commented on src/main/java/bob/command/CreateEventCommand.java:
```diff
@@ -0,0 +1,73 @@
+                    "I'm sorry, the proper usage of the event bob.command is 'event <description> /from <start> /to <end>'. Please try again!");
+        }
+
+        String startDateStr = startEnd[0].trim();
```
Maybe could use a proper noun as variable name like startDateInput and endDateInput?

senshir commented on src/main/java/bob/command/DeleteCommand.java:
```diff
@@ -0,0 +1,46 @@
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException, IllegalCommandException {
+        if (userInput.length != 2) {
+            throw new IllegalCommandException(
+                    "I'm sorry, the proper usage of the delete bob.command is 'delete <index>'. Please try again!");
```
I think for error messages the user doesn't need to know that command is packaged in bob. We could just say "the proper usage of the delete command is 'delete <index>'.

senshir commented on src/main/java/bob/storage/Storage.java:
```diff
@@ -0,0 +1,95 @@
+        try {
+            lines = Files.readAllLines(path);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
Maybe can use a custom exception to print what the error is too

senshir commented on src/main/java/bob/storage/Storage.java:
```diff
@@ -0,0 +1,95 @@
+import java.util.List;
+
+public class Storage {
+    private final String filePath;
```
Could create a path variable when Storage is constructed so that you don't have to create it in every method.

senshir reviewed (commented):
Overall great job on the naming conventions. One thing maybe you could look at is the need to create a new file every time you introduce a new command. Maybe this can be more compact?

advaypakhale committed:
fix: Remove windows carriage returns

advaypakhale committed:
feat: Add testing

advaypakhale committed:
feat: Add find command

advaypakhale committed:
docs: Add javadocs

advaypakhale committed:
fix: Remove AI errors

advaypakhale committed:
feat: Enforce coding standard

advaypakhale committed:
Merge branch 'branch-A-CodingStandard'

advaypakhale committed:
Merge branch 'branch-Level-9'

advaypakhale committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/bob/parser/Parser.java
#	src/main/java/bob/storage/Storage.java
#	src/main/java/bob/task/Task.java
#	src/main/java/bob/task/TaskList.java

advaypakhale committed:
fix: Add getter lost during merge conflict

advaypakhale committed:
style: Implement checkstyle in build process

advaypakhale committed:
Merge branch 'branch-A-CheckStyle'

advaypakhale committed:
refactor: Separate bob core functionality from TUI component

advaypakhale committed:
feat: Add gui

advaypakhale committed:
style: Modify checkstyle rules

advaypakhale committed:
chore: Modify gitignore

advaypakhale committed:
Merge branch 'branch-Level-10'

advaypakhale committed:
chore: Update gitignore

advaypakhale committed:
Merge branch 'master' into branch-A-Assertions

advaypakhale committed:
feat: Add assertions

There are currently no assertions in the code to document important assumptions.

Let's modify the code to add assertions at certain critical points.

Using assertions is necessary during testing to check sanity of assumptions.

advaypakhale committed:
chore: Remove text ui testing

There is a folder dedicated to text-ui-testing, which was not being
used.

Let's remove it.

This code is dead code that should be removed as part of code quality
standards.

advaypakhale committed:
Merge pull request #1 from advaypakhale:branch-A-Assertions

Branch-A-Assertions

advaypakhale committed:
Merge pull request #2 from advaypakhale:branch-A-CodeQuality

Branch-A-CodeQuality

advaypakhale committed:
Merge branch 'master' of https://github.com/advaypakhale/ip

advaypakhale committed:
feat: Add CI for gradle build and test

advaypakhale committed:
Merge branch 'master' of https://github.com/advaypakhale/ip

advaypakhale committed:
feat: Add sorting logic to task list

advaypakhale committed:
Merge branch 'branch-C-Sort'

advaypakhale committed:
docs: Modify AI log

advaypakhale committed:
chore: Update gitignore

advaypakhale committed:
feat: Add title to scene

advaypakhale committed:
fix: Modify mark and unmark command output

advaypakhale committed:
docs: Add Ui.png

advaypakhale committed:
feat: Add tick marks for completed tasks

advaypakhale committed:
docs: Update README.md

advaypakhale committed:
docs: Update AI.md for week 6

advaypakhale committed:
docs: Fix README formatting

advaypakhale committed:
style: Fix formatting

advaypakhale committed:
style: Fix formatting

advaypakhale committed:
style: Fix formatting

advaypakhale committed:
style: Fix formatting

advaypakhale committed:
fix: Remove refactor errors

advaypakhale committed:
docs: Update readme

advaypakhale committed:
fix: Modify splash text formatting

advaypakhale committed:
fix: Remove splash text for GUI

advaypakhale committed:
style: Fix whitespaces

advaypakhale committed:
docs: Remove duke reference
